core/src/test/java/org/apache/calcite/test/CalciteAssert.java

```diff
@@ -1830,19 +1830,19 @@ public AssertQuery withRel(final Function<RelBuilder, RelNode> relFn) {
       return withHook(Hook.STRING_TO_QUERY,
           (Consumer<Pair<FrameworkConfig, Holder<CalcitePrepare.Query>>>)
           pair -> {
-            final FrameworkConfig config = forceDecorrelate(pair.left);
+            final FrameworkConfig config = withDecorrelate(pair.left, false);
             final RelBuilder b = RelBuilder.create(config);
             pair.right.set(CalcitePrepare.Query.of(relFn.apply(b)));
           });
     }
 
     /** Creates a {@link FrameworkConfig} that does not decorrelate. */
-    private FrameworkConfig forceDecorrelate(FrameworkConfig config) {
+    private FrameworkConfig withDecorrelate(FrameworkConfig config, boolean needDecorrelate) {
       return Frameworks.newConfigBuilder(config)
```

Comment:
How about just remove the method and inline the code, the method name is mis-leading.

Reply:
> How about just remove the method and inline the code, the method name is mis-leading.

👌

Reply:
> How about just remove the method and inline the code, the method name is mis-leading.

Revised